prepare_data/src/main/groovy/org/orbisgis/osm/FormattingForAbstractModel.groovy

```diff
@@ -0,0 +1,369 @@
+        if ((r_height != null && r_height.isFloat()) || (b_r_height != null && b_r_height.isFloat())) {
+            if (b_height != null && b_height.isFloat()) {
+                if (b_r_height != null && b_r_height.isFloat()) {
+                    result = b_height.toFloat() - b_r_height.toFloat()
+                } else {
+                    result = b_height.toFloat() - r_height.toFloat()
+                }
+            } else {
+                if (b_r_height != null && b_r_height.isFloat()) {
+                    result = height.toFloat() - b_r_height.toFloat()
+                } else {
+                    result = height.toFloat() - r_height.toFloat()
+                }
+            }
+        }
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column height_roof according to the values of height and b_height
+ * @param row The row of the raw table to examine
+ * @return The calculated value of height_roof (default value : 0)
+ */
+static float getHeightRoof(def row) {
+    String height = row.getString("height")
+    String b_height = row.getString("b_height")
+    Float result = 0
+    if ((height != null && height.isFloat()) || (b_height != null && b_height.isFloat())) {
+        if (height != null && height.isFloat()) {
+            result = height.toFloat()
+        } else {
+            result = b_height.toFloat()
+        }
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column nb_lev according to the values of b_lev, r_lev and b_r_lev
+ * @param row The row of the raw table to examine
+ * @return The calculated value of nb_lev (default value : 0)
+ */
+static int getNbLevels (def row) {
+    String b_lev = row.getString("b_lev")
+    String r_lev = row.getString("r_lev")
+    String b_r_lev = row.getString("b_r_lev")
+    int result = 0
+    if (b_lev != null && b_lev.isFloat()) {
+        if ((r_lev != null && r_lev.isFloat()) || (b_r_lev != null && b_r_lev.isFloat())) {
+            if (r_lev.isFloat()) {
+                result = b_lev.toFloat() + r_lev.toFloat()
+            } else {
+                result = b_lev.toFloat() + b_r_lev.toFloat()
+            }
+        } else {
+            result = b_lev.toFloat()
+        }
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column width according to the value of width from OSM
+ * @param width The original width value
+ * @return the calculated value of width (default value : null)
+ */
+static Float getWidth (String width){
+    Float result
+    if (width != null && width.isFloat()) {
+        result=width.toFloat()
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column zindex according to the value of zindex from OSM
+ * @param width The original zindex value
+ * @return The calculated value of zindex (default value : null)
+ */
+static Integer getZIndex (String zindex){
+    Integer result
```

Comment:
The method body can be replaced by a ternary operator:
``` groovy
return (zindex != null && zindex.isInteger()) ? zindex.toInteger() : 0
```

Reply:
Done